sdk/cmake/host-tools.cmake

```
@@ -26,6 +26,48 @@ function(setup_host_tools)
         set(HOST_MODULE_SUFFIX ".so")
     endif()
 
+    # Normalize to the same format as our own ARCH, and add one for the VC shell
+    string(TOLOWER "${CMAKE_HOST_SYSTEM_PROCESSOR}" lowercase_CMAKE_HOST_SYSTEM_PROCESSOR)
+    if(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL x86_64 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL amd64)
+        set(HOST_ARCH amd64)
+        set(VCVARSALL_ARCH x86) # host-tools are not happy compiling for x64...
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL i686 OR lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL i386)
+        set(HOST_ARCH i386)
+        set(VCVARSALL_ARCH x86)
+    elseif(lowercase_CMAKE_HOST_SYSTEM_PROCESSOR STREQUAL arm)
+        set(HOST_ARCH arm)
+        set(VCVARSALL_ARCH arm)
+    else()
+        message(FATAL_ERROR "Unknown host architecture: ${lowercase_CMAKE_HOST_SYSTEM_PROCESSOR}")
+    endif()
+
+    if(ARCH STREQUAL HOST_ARCH)
+        set(HOST_TOOLS_CMAKE_COMMAND "${CMAKE_COMMAND}")
+        message("Not cross-compiling, no special host-tools cmake command")
+    elseif(MSVC)
+        message("Compiling on ${HOST_ARCH} for ${ARCH} (MSVC)")
+        set(HOST_TOOLS_CMAKE_COMMAND "${REACTOS_BINARY_DIR}/host-tools/cmake_shim.cmd")
+        if(MSVC_VERSION EQUAL 1900)
+            file(WRITE ${HOST_TOOLS_CMAKE_COMMAND}
+                "@call \"$ENV{VCINSTALLDIR}\\vcvarsall.bat\" ${VCVARSALL_ARCH}\n"
+                "\"${CMAKE_COMMAND}\" %*"
+            )
+        elseif(MSVC_VERSION GREATER_EQUAL 1910)
+            # 2017 and 2019 use the same folder structure
+            file(WRITE ${HOST_TOOLS_CMAKE_COMMAND}
+                "@set VSCMD_ARG_no_logo=1\n"
+                "@call \"$ENV{VCINSTALLDIR}\\Auxiliary\\Build\\vcvarsall.bat\" /clean_env\n"
+                "@call \"$ENV{VCINSTALLDIR}\\Auxiliary\\Build\\vcvarsall.bat\" ${VCVARSALL_ARCH}\n"
```

```suggestion
                "@call \"$ENV{VCINSTALLDIR}\\Auxiliary\\Build\\vcvarsall.bat\" ${CMAKE_HOST_SYSTEM_PROCESSOR}_${VCVARSALL_ARCH}\n"
```
shouldn't it be like that?

---

`amd64_x86` is also fine